modules/oe_newsroom_newsletter/src/Plugin/Block/NewsletterSubscriptionBlock.php

```diff
@@ -0,0 +1,192 @@
+    ];
+    $form['intro_text'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Introduction text'),
+      '#description' => $this->t('Text which will show on top of the form'),
+      '#maxlength' => 128,
+      '#default_value' => $config['intro_text'] ?? '',
+      '#required' => TRUE,
+    ];
+    $form['successful_subscription_message'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Successful subscription message'),
+      '#description' => $this->t('Text which will shown if the user successfully subscribed to the newsletters. Leave empty to use the message returned by the Newsroom API.'),
+      '#maxlength' => 128,
+      '#default_value' => $config['successful_subscription_message'] ?? '',
+    ];
+    $form['distribution_lists'] = [
+      '#type' => 'multivalue',
+      '#title' => $this->t('Newsletter distribution lists'),
+      '#description' => $this->t("If there's a single choice here, it will remain hidden on the (un)subscription form."),
+      '#cardinality' => 5,
+      '#required' => TRUE,
+      'sv_id' => [
+        '#type' => 'textfield',
+        '#title' => $this->t('Sv IDs'),
+        '#description' => $this->t('Comma-separated list of newsletter/distribution list IDs.'),
+        '#maxlength' => 128,
+      ],
+      'name' => [
+        '#type' => 'textfield',
+        '#title' => $this->t('Name of the distribution list'),
+        '#description' => $this->t('This is used to help the user identify which list they want to subscribe to.'),
+        '#maxlength' => 128,
+      ],
+      '#default_value' => $config['distribution_lists'] ?? [],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockValidate($form, FormStateInterface $form_state) {
+    parent::blockValidate($form, $form_state);
+
+    if (!empty($form_state->getValue('newsletters_language')) && !in_array($form_state->getValue('newsletters_language_default'), $form_state->getValue('newsletters_language'))) {
+      $form_state->setError($form['newsletters_language_default'], $this->t('The default language should be part of the possible newsletter languages.'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    parent::blockSubmit($form, $form_state);
+
+    $this->configuration['newsletters_language'] = $form_state->getValue('newsletters_language') ?? [];
+    $this->configuration['newsletters_language_default'] = $form_state->getValue('newsletters_language_default') ?: '';
+    $this->configuration['distribution_lists'] = $form_state->getValue('distribution_lists');
+    $this->configuration['intro_text'] = $form_state->getValue('intro_text');
+    $this->configuration['successful_subscription_message'] = $form_state->getValue('successful_subscription_message');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    if (!$this->newsroomClient->isConfigured()) {
```

Comment:
This can be moved in the blockAccess() method, so we avoid any rendering if the client is not configured for usage.